include/highfive/bits/H5Converter_misc.hpp

```diff
@@ -174,6 +175,34 @@ struct data_converter<
     size_t _dim;
 };
 
+// apply conversion to std::array
+template <typename T, std::size_t S>
+struct data_converter<
+    std::array<T, S>,
+    typename std::enable_if<(
+                         std::is_same<T, typename type_of_array<T>::type>::value)>::type> {
+    inline data_converter(std::array<T,S>& vec, DataSpace& space, size_t dim = 0)
+    : _space(&space), _dim(dim) {
+        assert(_space->getDimensions().size() >= dim);
+        (void)vec;
```

Comment:
All other types are resizing the back storage according to `space` except in the case of plain C arrays, where not even the size can be checked. Since here you can't resize, but at least you can check the size, you should verify that space.getDimensions()[0] == S and throw DataSpaceException otherwise. Exercise this in the test.

Reply:
Changed, see if this is better.

Reply:
You don't need to use 'at' if you have already checked that the size is not 0. But what I was suggesting is to have an if and throw an exception. In H5Slice_traits_misc.hpp you can see that the dimensions are checked and an exception is if they don't match. I don't see a good reason to not doing the same thing here as the penalty of checking that compared to the IO operation is going to be negligible. I've realized that the assert is not needed because of the check made by the caller, but I know the other converters are doing it.

Reply:
Is the new version better?